packages/smooth_app/lib/database/dao_secured_string.dart

```diff
@@ -0,0 +1,66 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:smooth_app/database/abstract_dao.dart';
+import 'package:smooth_app/database/local_database.dart';
+
+enum SecuredValues {
+  USER_ID,
+  PASSWORD,
+}
+
+/// Where we store extra secured strings
+class DaoSecuredString extends AbstractDao {
+  DaoSecuredString(final LocalDatabase localDatabase) : super(localDatabase);
+
+  static const String _hiveBoxName = 'credentialsBox';
+  //The key-value key under which the base64 encrypted encryption key is stored
+  static const String _encryptionKeyKey = 'securedStringsKey';
+
+  @override
+  Future<void> init() async {
+    const FlutterSecureStorage secureStorage = FlutterSecureStorage();
```

Comment:
I know you just copied code from https://docs.hivedb.dev/#/advanced/encrypted_box (by the way you should have added the reference in the comments): I cannot blame you but that goes too far IMHO. Let me explain.
You use an additional package, `flutter_secure_storage`, to store "securely" a generated key. Then you use this key to encrypt a hive box, in order to get a couple of values.
Assuming `flutter_secure_storage` is secure, you'd be better off storing login and password directly in it, wouldn't you? It's not like we're specifically needing hive here.
What do you think?

Reply:
Actually makes sense since we don't have so much data to be securely stored. Should it then still be named DaoSecuredBox though

Reply:
Probably some misunderstanding here...

I don't get it: the initial goal is to store securely the password (and if you want, the login).
`flutter_secure_storage` does that directly and perfectly, doesn't it? There's no added value in `hive` here.

Using `hive` here adds a layer of complexity in the init phase; it's not like we want all the hive database to be encrypted because we store tons of confidential data. The only confidential data are the password and the login.

Reply:
Yes I understand what you mean, the question was whether we should keep DaoSecuredString but without Hive code inside it or to put the new code in for example the UserManagmentHelper

Reply:
It shouldn't be a problem to name a class `DaoSecuredString` that does not rely on `hive` but just on `flutter_secure_storage`. After all, the goal of a DAO is to hide the database and the complexity: "just give me access to the data!"